### PR TITLE
Introduce methods for getting all scheduled emails

### PR DESCRIPTION
**Pull request by scavnickyj**
JIRA: TNT-511

---

Supported PR commands:

| Command                  | Description            |
| ------------------------ | ---------------------- |
| `ok to test`             | Re-run standard checks |
| `extended test`          | BackstopJS tests       |
| `extended check sonar`   | SonarQube tests        |
| `extended check cypress` | Cypress E2E tests      |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `check-extended` passes
-   [ ] `check-extended-cypress` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/7218/ : SUCCESS in 5s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/9065/ : SUCCESS in 15m 30s

File: libs/sdk-backend-bear/src/backend/workspace/dashboards/index.ts [hunk 1]
```diff
@@ -305,14 +306,26 @@ export class BearWorkspaceDashboards implements IWorkspaceDashboardsService {
         return toSdkModel.convertScheduledMail(createdBearScheduledMail) as IScheduledMail;
     };
 
+    public getScheduledMailsForDashboard = async (
+        dashboardRef: ObjRef,
+        createdByCurrentUser = false,
```
Comment (kandl) [inline libs/sdk-backend-bear/src/backend/workspace/dashboards/index.ts:311]: We should likely hide `createdByCurrentUser` flag under some options object (so if we introduce more options in the future, eg for some other filtering, we can avoid parameters hell and breaking changes)

Comment (scavnickyj) [reply libs/sdk-backend-bear/src/backend/workspace/dashboards/index.ts]: Good point, fixed

**Review by ivanmjartan — APPROVED** [review]

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR2329

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/2087/ : SUCCESS in 1m 11s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/2371/ : SUCCESS in 19s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/1190/ : SUCCESS in 14m 20s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/9076/ : SUCCESS in 15m 36s

**Pull request merged**